File: Chapter 05/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter05/section_5_6_crypto_tracker/before/AttemptedTest.java
package com.grokkingsoftwarearchitecture.chapter05.section_5_6_crypto_tracker.before;

/**
 * A demonstration of why Tight Coupling ruins testability.
 */
public class AttemptedTest {
    public static void run() {
        System.out.println("\n--- ATTEMPTING TO TEST (BEFORE) ---");
        
        PortfolioManager manager = new PortfolioManager();

        System.out.println("Test Action: Calculating value of 1 BTC...");
        
        try {
            double value = manager.calculateTotalValue(1.0);
            
            // ASSERT
            // We cannot assert equality because the price changes constantly.
            System.out.println("Result: " + value);
            System.out.println("FAIL: This test is FLAKY. We cannot assert a fixed price.");
        } catch (Exception e) {
            System.out.println("CRASH: Test failed completely. No internet connection.");
        }
    }
}

File: Chapter 05/JS/section_5_6_crypto_tracker/before/attemptedTest.js
const PortfolioManager = require('./portfolioManager');

/**
 * A demonstration of why Tight Coupling ruins testability.
 */
class AttemptedTest {
    static async run() {
        console.log("\n--- ATTEMPTING TO TEST (BEFORE) ---");
    
    const btcAmount = 2.5; // Using a non-integer to prove it handles decimals
    const monitor = new PortfolioManager();
    const usd = new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' });

    console.log(`Test Action: Verifying portfolio for ${btcAmount} BTC...`);
    
    const result = monitor.calculateTotalValue(btcAmount);
    
    // NEW: Showing the math in the test output
    console.log(`----------------------------------------`);
    console.log(`Quantity: ${btcAmount} BTC`);
    console.log(`Market Price: ${usd.format(result / btcAmount)}`);
    console.log(`Total Value: ${usd.format(result)}`);
    console.log(`----------------------------------------`);

    if (result > 0) {
        console.log("SUCCESS: Portfolio calculation verified.");
    } else {
        console.error("FAIL: Calculation returned zero or invalid data.");
    }
}}

module.exports = AttemptedTest;